sterror/sterror.go

```diff
@@ -0,0 +1,157 @@
+package sterror
+
+import (
+	"errors"
+)
+
+type Op string
+
+func (o Op) String() string {
+	return string(o)
+}
+
+type Scope string
+
+func (s Scope) String() string {
+	return string(s)
+}
+
+type Info string
+
+func (i Info) String() string {
+	return string(i)
+}
+
+const (
+	colon     = ": "
+	hyphen    = " - "
+	separator = "\n"
+)
+
+const (
+	Undefined Scope = "Undefined"
+	Host      Scope = "Host"
+	Network   Scope = "Network"
+	Opts      Scope = "Opts"
+	Ospkg     Scope = "Ospkg"
+	Stlog     Scope = "Stlog"
+	Trust     Scope = "Trust"
+)
+
+/* Error-Wrapper
+- op: Function that raised the error
+- Scope: Package that raised the error
+- Info: additional Info passed as a string
+- Err: Wrapped underlying error.
+*/
+type Error struct {
+	Op    Op
+	Scope Scope
+	Info  Info
+	Err   error
+}
+
+// returns the error as a string.
+func (e Error) Error() string {
+	composedErrorString := e.Scope.String()
+
+	switch {
+	case e.Op != "" && e.Info != "":
+		composedErrorString += colon + e.Op.String() + hyphen + e.Info.String()
+	case e.Op != "":
+		composedErrorString += colon + e.Op.String()
+	case e.Info != "":
+		composedErrorString += colon + e.Info.String()
+	}
+
+	if e.Err != nil {
+		composedErrorString += separator + e.Err.Error()
+	}
+
+	return composedErrorString
+}
+
+// create a new Error Wrapper.
+func NewError(args ...interface{}) Error {
+	err := Error{}
+	err.Scope = Undefined
+
+	if len(args) == 0 {
+		return err
+	}
+
+	for _, arg := range args {
+		switch arg := arg.(type) {
+		case Op:
+			err.Op = arg
+		case Scope:
+			err.Scope = arg
+		case Info:
+			err.Info = arg
+		case error:
+			err.Err = arg
+		default:
```

Comment:
We could try to catch other types by something like:
```
default:
    err.Info = err.foo + fmt.Sprintf("%v", arg)
````
But thats  something to discuss.. mayby just leaf it empty or make a call to stlog.Debug

Reply:
Then lets stick to doing nothing here.